src/WebJobs.Script.WebHost/FileMonitoringService.cs

```diff
@@ -411,6 +424,15 @@ public void Dispose()
                 {
                     Interlocked.Decrement(ref _fileMonitoringService._suspensionRequestsCount);
                     _fileMonitoringService._typedLogger.LogDebug($"Exiting restart suspension scope. ({_fileMonitoringService._suspensionRequestsCount} requests).");
+                    if (_autoResume)
```

Comment:
_suspensionRequestsCount might still be > 0 here right? Seems like you need to also check this value and only restart if it's now zero. E.g. you could have nested using blocks.

Reply:
Please add a test for that scenario - nested restart disposal blocks

Reply:
We will check later `_suspensionRequestsCount` here:
https://github.com/Azure/azure-functions-host/blob/alrod/fixrestart/src/WebJobs.Script.WebHost/FileMonitoringService.cs#L131

Reply:
Added the test for nested disposal blocks:
https://github.com/Azure/azure-functions-host/pull/7086/commits/8aefaff68c037a41b059959844baef47db59aa8c